pinot-broker/src/main/java/com/linkedin/pinot/broker/broker/BrokerServerBuilder.java

```diff
@@ -107,81 +65,29 @@
-    TransportClientConf conf = new TransportClientConf();
-    conf.init(transportConfigs);
-
-    _registry = new MetricsRegistry();
-    MetricsHelper.initializeMetrics(_config.subset(METRICS_CONFIG_PREFIX));
-    MetricsHelper.registerMetricsRegistry(_registry);
-    _brokerMetrics = new BrokerMetrics(_registry, !emitTableLevelMetrics());
-    _brokerMetrics.initializeGlobalMeters();
     _state.set(State.INIT);
-    _eventLoopGroup = new NioEventLoopGroup();
-    /**
-     * Some of the client metrics uses histogram which is doing synchronous operation.
-     * These are fixed overhead per request/response.
-     * TODO: Measure the overhead of this.
-     */
-    final NettyClientMetrics clientMetrics = new NettyClientMetrics(_registry, "client_");
-
-    // Setup Netty Connection Pool
-    _resourceManager = new PooledNettyClientResourceManager(_eventLoopGroup, new HashedWheelTimer(), clientMetrics);
-    _poolTimeoutExecutor = new ScheduledThreadPoolExecutor(50);
-    // _requestSenderPool = MoreExecutors.sameThreadExecutor();
-
-    _requestSenderPool = Executors.newCachedThreadPool();
-
-    final ConnectionPoolConfig connPoolCfg = conf.getConnPool();
-
-    _connPool = new KeyedPoolImpl<PooledNettyClientResourceManager.PooledClientConnection>(connPoolCfg.getMinConnectionsPerServer(),
-        connPoolCfg.getMaxConnectionsPerServer(), connPoolCfg.getIdleTimeoutMs(), connPoolCfg.getMaxBacklogPerServer(),
-        _resourceManager, _poolTimeoutExecutor, _requestSenderPool, _registry);
-    // MoreExecutors.sameThreadExecutor(), _registry);
-    _resourceManager.setPool(_connPool);
-
-    // Setup Routing Table
-    if (conf.getRoutingMode() == RoutingMode.CONFIG) {
-      final CfgBasedRouting rt = new CfgBasedRouting();
-      rt.init(conf.getCfgBasedRouting());
-      _routingTable = rt;
-    } else {
-      // Helix based routing is already initialized.
-    }
-
-    // Setup ScatterGather
-    _scatterGather = new ScatterGatherImpl(_connPool, _requestSenderPool);
-
-    // Setup the broker pruner service
-    String[] prunerNames = _config.getStringArray(BROKER_SEGMENT_PRUNERS);
-    if (prunerNames == null) {
-      prunerNames = DEFAULT_BROKER_SEGMENT_PRUNERS;
-    }
-    SegmentZKMetadataPrunerService brokerPrunerService = new SegmentZKMetadataPrunerService(prunerNames);
-
-    // Setup Broker Request Handler
-    ReduceServiceRegistry reduceServiceRegistry = buildReduceServiceRegistry();
-    _accessControlFactory = AccessControlFactory.loadFactory(_config.subset(BROKER_ACCESS_CONTROL_PREFIX));
-    _requestHandler = new BrokerRequestHandler(_routingTable, _timeBoundaryService, _scatterGather,
-        reduceServiceRegistry, brokerPrunerService, _brokerMetrics, _config, _accessControlFactory,
-        _tableQueryQuotaManager);
+    _brokerRequestHandler =
+        new ConnectionPoolBrokerRequestHandler(_config, _routingTable, _timeBoundaryService, _accessControlFactory,
```

Comment:
What is the plan for the migration after you implement async request handler? After your refactoring, we will have multiple implementations of broker request handlers. We may need to consider to add the configuration and use the factory to initialize broker request handler (and use async one by default).

Reply:
That will be in the PR of async request handler